drivers/sensor/adi/adxl372/adxl372.c

```diff
@@ -364,7 +364,16 @@ int adxl372_get_status(const struct device *dev,
 	}
 
 	if (fifo_entries) {
-		length += 2U;
+		if (status2) {
+			length += 2U;
+		} else {
+			/* Registers STATUS1, STATUS2, FIFO_ENTRIES2 and FIFO_ENTRIES1
+			 * are one after the other. If user wants fifo_entries and
+			 * not status2, to get the correct values, STATUS2 register
+			 * also must be read but read value will be ignored.
+			 */
+			length += 3U;
+		}
```

Comment:
I'm not sure I understand the rationale for having a variable-length transfer here.

Wouldn't it be just easier to always pull the 4-bytes and depending on whether the pointer is valid: pass that config over?

Reply:
My rationale was that I wanted to reduce SPI overhead and not read registers that are not necessary (FIFO ENTRIES1/2).

Reply:
Ok - I personally find it harder to maintain this way vs the overhead of 1 or 2 extra bytes. It's a subjective thought though. Approving.